src/gui/MainWindow.hpp

```diff
@@ -41,7 +41,7 @@ class MainWindow : public QMainWindow
     QString m_importFilePath;
 
     std::unique_ptr<ModbusCom> m_modbus;
-    std::vector<std::shared_ptr<MdiChild>> m_mdiChilds;
+    std::vector<MdiChild *> m_mdiChilds;
```

Comment:
Raw Pointer?

Reply:
Unfortunately, there is no other way. Using smart pointers will cause multiple deallocations, from the smart pointer itself and from the Qt memory management, ending in segmentation faults.
In other words, this vector of pointers will be deallocated upon the MainWindow destruction.

Reply:
Open a bug to Qt